qadence/ml_tools/config.py

```diff
@@ -28,6 +29,80 @@
+    an OptimizeResult instance.
+
+    Attributes:
+        callback (Callable[..., None]): Callback function accepting an
+            OptimizeResult as ifrst argument.
+        callback_condition (Callable[..., bool] | None, optional): Function that conditions the
+            call to callback. Defaults to None.
+        every (int, optional): Callback to be called each `every` epoch. Defaults to 1.
+            If callback_condition is None, we set
+            callback_condition to returns True when iteration % every == 0.
+        call_before_opt (bool, optional): If true, callback is applied before training.
+            Defaults to False.
+        call_end_epoch (bool, optional): If true, callback is applied during training,
+            after an epoch is performed. Defaults to True.
+        call_after_opt (bool, optional): If true, callback is applied after training.
+            Defaults to False.
+        call_during_eval (bool, optional): If true, callback is applied during evaluation.
+            Defaults to False.
+    """
+
+    def __init__(
+        self,
+        callback: Callable[..., None],
+        callback_condition: Callable[..., bool] | None = None,
+        every: int = 1,
+        call_before_opt: bool = False,
+        call_end_epoch: bool = True,
+        call_after_opt: bool = False,
+        call_during_eval: bool = False,
+    ) -> None:
+        """Initialized Callback.
+
+        Args:
+            callback (Callable[..., None]): Callback function accepting an
+                OptimizeResult as ifrst argument.
+            callback_condition (Callable[..., bool] | None, optional): Function that conditions the
+                call to callback. Defaults to None.
+            every (int, optional): Callback to be called each `every` epoch. Defaults to 1.
+                If callback_condition is None, we set
+                callback_condition to returns True when iteration % every == 0.
+            call_before_opt (bool, optional): If true, callback is applied before training.
+                Defaults to False.
+            call_end_epoch (bool, optional): If true, callback is applied during training,
+                after an epoch is performed. Defaults to True.
+            call_after_opt (bool, optional): If true, callback is applied after training.
+                Defaults to False.
+            call_during_eval (bool, optional): If true, callback is applied during evaluation.
+                Defaults to False.
+        """
+        self.callback = callback
+        self.call_before_opt = call_before_opt
+        self.call_end_epoch = call_end_epoch
+        self.call_after_opt = call_after_opt
+        self.call_during_eval = call_during_eval
+
+        self.every = every
+
+        if callback_condition is None:
+            if self.every > 1:
+                self.callback_condition = lambda opt_result: opt_result.iteration % self.every == 0
```

Comment:
This will always be true for self.every==1, but I guess running it each time unnecessary would be costing extra.